sockeye/utils.py

```diff
@@ -30,13 +30,57 @@
 import mxnet as mx
 import numpy as np
 
+from sockeye import __version__
+
 logger = logging.getLogger(__name__)
 
 
 class SockeyeError(Exception):
     pass
 
 
+def check_version(version: str):
+    """
+    Checks given version against code version and determines compatibility.
+    Throws if versions are incompatible.
+
+    :param version: Given version.
+    """
+    code_version = parse_version(__version__)
+    given_version = parse_version(version)
+    check_condition(code_version[0] == given_version[0],
+                    "Given release version (%s) does not match release code version (%s)" % (version, __version__))
+    check_condition(code_version[1] == given_version[1],
+                    "Given major version (%s) does not match major code version (%s)" % (version, __version__))
+    if code_version[2] != given_version[2]:
+        logger.warning("Given minor version (%s) does not match code version (%s)", version, __version__)
```

Comment:
I think some indication in the log is useful. Do you want to reduce to logger.info or what do you mean by "just do a logger.warning" because that is already the case, right?